R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,236 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      summarise_at(vars(avg), mean, na.rm = TRUE) %>%
+      rename(FodderHerb = avg) %>%
+      # Note: Setting FodderGrass price as a ratio to FodderHerb
+      mutate(FodderGrass = FodderHerb * PRICERATIO_GRASS_ALFALFA) %>%
+      # NOTE: Setting Pasture price equal to FodderGrass price
+      mutate(Pasture = FodderGrass) %>%
+      gather(GCAM_commodity, Price_USDt) %>%
+      # Combine crop commodities to get all primary agricultural commodities in a single tibble
+      bind_rows(Price_ag) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDt * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+      select(-Price_USDt) %>%
+      # Combine animal products
+      bind_rows(Price_an) %>%
+      mutate(unit = "1975$/kg") ->
+      Price_ag_an
+
+    # Part 3: Forest products, use FAO forest export value and export quantities to estimate price
+    FAO_USA_For_Exp_t_USD_FORESTAT %>%
+      select(-countries, -country.codes, -item, -item.codes, -element.codes) %>%
+      gather(year, value, -element) %>%
+      # Calculate average value and quantity over price years
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(element) %>%
+      summarise_at(vars(value), mean, na.rm = TRUE) %>%
+      ungroup() %>%
+      # Modity element names to one word so that they can be used as column names when spreading element and doing calculations
+      mutate(element = if_else(element == "Export Quantity (m3)", "Exp_m3", "ExpV_USD")) %>%
+      mutate(GCAM_commodity = "Forest") %>%
+      spread(element, value) %>%
+      # Calculate forest price as export value (in thous USD) divided by export quantity
+      mutate(Price_USDm3 = ExpV_USD * 1000 / Exp_m3) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDm3 * CONV_2004_1975_USD, digits = DIGITS_CALPRICE)) %>%
+      select(GCAM_commodity, calPrice) %>%
+      mutate(unit = "1975$/m3") %>%
+      # Part 4: merging everything into a single table
+      bind_rows(Price_ag_an) ->
+      Price_ag_an_For
+
+    # Check that all commodities being modeled have prices assigned
+    FAO_ag_items_PRODSTAT %>%
+      select(GCAM_commodity) %>%
+      distinct() %>%
+      bind_rows(distinct(select(FAO_an_items_PRODSTAT, GCAM_commodity))) %>%
+      na.omit() %>%
+      # Convert tibble to vector
+      collect() %>%
+      .[['GCAM_commodity']] -> commodities
+    Price_ag_an_For %>%
+      collect() %>%
+      # Convert tibble to vector
+      .[['GCAM_commodity']] -> prices
+    # Check missing commodities
+    if(any(!(commodities %in% prices))){
+      missing_commodities <- commodities[!(commodities %in% prices)]
+      stop("Missing commodity prices for ", missing_commodities)
+    }
+
+    # Animal products prices are in USD per metric tonne, so the CONV_T_METRIC_SHORT in the old data system is unnecessary and dropped.
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
```

Comment:
If I'm understanding it correctly, we only put the fix in the `else{}`, but it does not affect the output. The chunk still duplicates the exact old behavior and produces the same old output, so that the test succeeds. Correct?